File: src/main/java/hwr/oop/examples/classes/methods/Train.java
package hwr.oop.examples.classes.methods;

class Train {
    private final int numberOfRailCars;
    private int speed;

    public Train(int numberOfRailCars) {
        this.numberOfRailCars = numberOfRailCars;
    }

    public void accelerateTo(int speed) {
        this.speed = speed;
    }

    @Override
    public String toString() {
        return "Train{" +
                "numberOfRailCars=" + numberOfRailCars +
                ", speed=" + speed +
                '}';
    }
}

File: examples-kotlin/lecture-02/src/main/kotlin/hwr/oop/examples/visibility_modifiers/Train.kt
package hwr.oop.examples.visibility_modifiers

open class Train {
  private var speed: Int = 0
  private var engineStarted: Boolean = false

  fun isEngineStarted(): Boolean {
    return engineStarted
  }

  internal fun startEngine() {
    engineStarted = true
    println("Engine started")
  }

  open fun accelerateTo(speed: Int) {
    if (!engineStarted) {
      startEngine()
    }
    this.speed = speed
  }
}